examples/conservative-raster/main.rs

```diff
@@ -0,0 +1,333 @@
+            let bind_group_layout =
+                device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+                    label: Some("upscale bindgroup"),
+                    entries: &[
+                        wgpu::BindGroupLayoutEntry {
+                            binding: 0,
+                            visibility: wgpu::ShaderStage::FRAGMENT,
+                            ty: wgpu::BindingType::Texture {
+                                sample_type: wgpu::TextureSampleType::Float { filterable: false },
+                                view_dimension: wgpu::TextureViewDimension::D2,
+                                multisampled: false,
+                            },
+                            count: None,
+                        },
+                        wgpu::BindGroupLayoutEntry {
+                            binding: 1,
+                            visibility: wgpu::ShaderStage::FRAGMENT,
+                            ty: wgpu::BindingType::Sampler {
+                                filtering: false,
+                                comparison: false,
+                            },
+                            count: None,
+                        },
+                    ],
+                });
+
+            let pipeline_layout = device.create_pipeline_layout(&wgpu::PipelineLayoutDescriptor {
+                label: None,
+                bind_group_layouts: &[&bind_group_layout],
+                push_constant_ranges: &[],
+            });
+            let shader = device.create_shader_module(&wgpu::ShaderModuleDescriptor {
+                label: None,
+                source: wgpu::ShaderSource::Wgsl(Cow::Borrowed(include_str!("upscale.wgsl"))),
+                flags: wgpu::ShaderFlags::all(),
+            });
+            (
+                device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+                    label: Some("Upscale"),
+                    layout: Some(&pipeline_layout),
+                    vertex: wgpu::VertexState {
+                        module: &shader,
+                        entry_point: "vs_main",
+                        buffers: &[],
+                    },
+                    fragment: Some(wgpu::FragmentState {
+                        module: &shader,
+                        entry_point: "fs_main",
+                        targets: &[sc_desc.format.into()],
+                    }),
+                    primitive: wgpu::PrimitiveState::default(),
+                    depth_stencil: None,
+                    multisample: wgpu::MultisampleState::default(),
+                }),
+                bind_group_layout,
+            )
+        };
+
+        let (low_res_target, bind_group_upscale) =
+            Self::create_low_res_target(sc_desc, device, &bind_group_layout_upscale);
+
+        println!("------------------------------------------------------------------------");
+        println!("Press Space to toggle between regular and conservative rasterization");
```

Comment:
Would it be easier to just draw both conservative, non-conservative (with different color), and lines on top, to show the difference? I think non-interactive examples are easier to test.

Reply:
![image](https://user-images.githubusercontent.com/1220815/111223978-254d7800-85de-11eb-8f5b-7cd1bf72bb97.png)
tried it here. Don't like it as much but I get your point - much easier to go through them and maybe even implement an image comparison test later.

Reply:
looks great!

Reply:
okay. I hate the colors, but it will do :D

Reply:
we can change the colors :)